mllib-dal/pom.xml

```diff
@@ -16,7 +16,6 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <scala.version>2.12.10</scala.version>
         <scala.binary.version>2.12</scala.binary.version>
```

Comment:
did you check if we need to revise scala version for each spark versions?

Reply:
From [spark doc](https://spark.apache.org/docs/latest/),
"For the Scala API, Spark 3.1.1 uses Scala 2.12. You will need to use a compatible Scala version (2.12.x)."
So suppose 2.12 is the right scala version.